genai-perf/genai_perf/config/generate/perf_analyzer_config.py

```diff
@@ -246,12 +246,16 @@ def _add_required_args(self, config: ConfigCommand) -> List[str]:
     def _add_perf_analyzer_args(self, config: ConfigCommand) -> List[str]:
         perf_analyzer_args = []
 
+        if config.perf_analyzer.warmup_request_count > 0:
+            perf_analyzer_args += [
+                "--warmup-request-count",
+                f"{config.perf_analyzer.warmup_request_count}",
+            ]
+
         if config.input.prompt_source != PromptSource.PAYLOAD:
             perf_analyzer_args += [
                 f"--stability-percentage",
                 f"{config.perf_analyzer.stability_percentage}",
-                f"--warmup-request-count",
-                f"{config.perf_analyzer.warmup_request_count}",
```

Comment:
Previously, the warmup feature wasn't supported in GAP "payload" mode, but now it is.

In the old code, warmup could only get appending to the PA command if "payload" mode _wasn't_ being used.

Now, valid warmup gets appending to PA command regardless of "payload" mode being used.